client/components/tinymce/plugins/mentions/mentions.jsx

```diff
@@ -0,0 +1,142 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import ReactDomServer from 'react-dom/server';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import SuggestionList from './suggestion-list';
+import EditorMention from './editor-mention';
+import QueryUsersSuggestions from 'components/data/query-users-suggestions';
+import { getUserSuggestions } from 'state/users/suggestions/selectors';
+
+const Mentions = React.createClass( {
+	displayName: 'Mentions',
+
+	propTypes: {
+		siteId: React.PropTypes.number,
+		editor: React.PropTypes.object,
+		suggestions: React.PropTypes.array
+	},
+
+	getDefaultProps() {
+		return {
+			suggestions: []
+		};
+	},
+
+	getInitialState() {
+		return {
+			query: '',
+			showPopover: false
+		};
+	},
+
+	componentDidMount: function() {
+		const { editor } = this.props;
+
+		editor.on( 'keyup', this.onKeyUp );
+	},
+
+	getCurrentText: function() {
+		const rng = this.props.editor.selection.getRng( 1 );
+
+		return rng.startContainer.textContent;
```

Comment:
Actually, this code was buggy so I've pushed a [new commit](https://goo.gl/9ILxHp). As part of that, `rng` has been renamed to `range` to make it more obvious what it is.

To your other point about deeply nested properties, `editor.selection` will return an instance of the `Selection` class, and `getRng()` will return the browser's internal range object. From there, the rest of the code (e.g. `range.startContainer.textContent` and `range.startOffset`) uses [native browser functionality](https://developer.mozilla.org/en-US/docs/Web/API/Range). Given this, do you still think there's cause for concern?